src/vm/eventreporter.cpp

```diff
@@ -343,6 +343,34 @@ void EventReporter::AddStackTrace(SString& s)
     }
 }
 
+//---------------------------------------------------------------------------------------
+//
+// Add the stack trace of exception passed to managed FailFast call (Environment.FailFast()) to Event Log
+//
+// Arguments:
+//    s       - String representation of the stack trace of argument exception
+//
+// Return Value:
+//    None.
+//
+void EventReporter::AddFailFastStackTrace(SString& s)
+{
+    CONTRACTL
+    {
+        THROWS;
+        GC_NOTRIGGER;
+        SO_INTOLERANT;
+        MODE_ANY;
+    }
+    CONTRACTL_END;
+
+    _ASSERTE(m_eventType == ERT_ManagedFailFast);
+
+    m_Description.Append(W("Exception stack:\n"));
```

Comment:
Should this be trying to load the resource (`.LoadResource(`) and falling back to hard coded if that fails, as the rest of hte file does?